integration-cli/docker_cli_diff_test.go

```diff
@@ -40,30 +41,42 @@ func (s *DockerSuite) TestDiffFilenameShownInOutput(c *check.C) {
 func (s *DockerSuite) TestDiffEnsureDockerinitFilesAreIgnored(c *check.C) {
 	// this is a list of files which shouldn't show up in `docker diff`
 	dockerinitFiles := []string{"/etc/resolv.conf", "/etc/hostname", "/etc/hosts", "/.dockerinit", "/.dockerenv"}
+	containerCount := 5
+
+	var wg sync.WaitGroup
+	wg.Add(containerCount)
 
 	// we might not run into this problem from the first run, so start a few containers
-	for i := 0; i < 20; i++ {
-		containerCmd := `echo foo > /root/bar`
-		runCmd := exec.Command(dockerBinary, "run", "-d", "busybox", "sh", "-c", containerCmd)
-		out, _, err := runCommandWithOutput(runCmd)
-		if err != nil {
-			c.Fatal(out, err)
-		}
+	for i := 0; i < containerCount; i++ {
+		//start tests concurrent
+		go func() {
 
-		cleanCID := strings.TrimSpace(out)
+			defer wg.Done()
 
-		diffCmd := exec.Command(dockerBinary, "diff", cleanCID)
-		out, _, err = runCommandWithOutput(diffCmd)
-		if err != nil {
-			c.Fatalf("failed to run diff: %s, %v", out, err)
-		}
+			containerCmd := `echo foo > /root/bar`
+			runCmd := exec.Command(dockerBinary, "run", "-d", "busybox", "sh", "-c", containerCmd)
+			out, _, err := runCommandWithOutput(runCmd)
+			if err != nil {
+				c.Fatal(out, err)
```

Comment:
I suppose you can just exit from the goroutine after sending the error to an error chan to be handled outside, wdyt?